### DISCUSSION TITLE
Client side Throughput Control

### DISCUSSION THREAD

**Discussion by Bru456** [Azure/azure-cosmos-dotnet-v3#4929] [Ideas]
Hi Team, 

I can see that the Java SDK supports `ThroughputControl` to allow clients to limit the amount of RUs that they consume. https://learn.microsoft.com/en-us/azure/cosmos-db/nosql/throughput-control-spark

I am wondering if this will be coming to the .Net SDK as I would like to limit the throughput of my clients to prevent them from consuming all available RUs. 

Say I have a background data refresh service that batch loads data into Cosmos. I would like to rate limit this to allow customer facing services to not be impacted by background services.

**Comment by Pilchie** [comment]
Not exactly the same thing, but the service has been working on a feature to allow "throughput buckets" that clients can assign requests to, and have the server enforce that buckets don't exceed a controlled limit.  That feature should be coming soon, and will be exposed to the .NET SDK as well.
